README.md

```diff
@@ -1,189 +1,248 @@
+The specified algorithm must be compatible with values of `private_key` and `sign_cert`.
 
-To create your own temporary files for testing you can execute the following command
+The key and cert can also be placed in the environment variables `C2PA_PRIVATE_KEY` and `C2PA_PUB_CERT`. These two variables are used to set the private key and public certificates. For example, to sign with es256 signatures using the content of a private key file and certificate file, you would run:
 
 ```shell
-sudo openssl req -new -newkey rsa:4096 -sigopt rsa_padding_mode:pss -days 180 -extensions v3_ca -addext "keyUsage = digitalSignature" -addext "extendedKeyUsage = emailProtection" -nodes -x509 -keyout private.key -out certs.pem -sha256
+set C2PA_PRIVATE_KEY=$(cat my_es256_private_key)
+set C2PA_PUB_CERT=$(cat my_es256_certs)
+```
+
+Both the `private_key` and `sign_cert` should be in PEM format. The `sign_cert` should contain a PEM certificate chain starting for the end-entity certificate used to sign the claim ending with the intermediate certificate before the root CA certificate. See the ["sample" folder](https://github.com/contentauth/c2patool/tree/main/sample) for example certificates.
+
+To create your own temporary files for testing, you can execute the following command:
+
+```shell
+sudo openssl req -new -newkey rsa:4096 
+   -sigopt rsa_padding_mode:pss \ 
+   -days 180 \
+   -extensions v3_ca \
+   -addext "keyUsage = digitalSignature" \
+   -addext "extendedKeyUsage = emailProtection" \
+   -nodes -x509 -keyout private.key -out certs.pem -sha256
 ```	
 
-Note you may have need to update your openssl version if the above command does not work.
+**Note:** You may have need to update your `openssl` version if the above command does not work. You will likely need version 3.0 or later. You can check the version that is installed by typing `openssl version`.
 
-c2patool can also timestamp the signature data that is embedded.  This is useful for validating an asset when the embedded 
-certificates have expired.  If the config has a ta_url set, c2patool will attempt to timestamp the signature using the TA service at the provided URL.  The TA must be RFC3161 compliant.  Example TA setting:
+c2patool can also timestamp the embedded signature data.  This is useful for validating an asset when the embedded certificates have expired. If the config has a `ta_url` set, c2patool will attempt to timestamp the signature using the TA service at the provided URL. The TA must be [RFC3161](https://datatracker.ietf.org/doc/html/rfc3161) compliant. For example,
 
-```ta_url=http://timestamp.digicert.com```
+```shell
+ta_url=http://timestamp.digicert.com
+```
 
-## Configuration file format
+##### Manifest definition format
 
-The Configuration file is a JSON formatted file with a .json extension:
+The manifest definition file is a JSON serialized file (`.json`). The schema for this type is as follows:
 
-The schema for this type is as follows:
 ```json
```

Comment:
@gpeacock Should this be a linked file in the repo instead of inline in the README?

Reply:
That's fine with me, probably cleaner that way and we can then validate the schema is correct.